public/javascripts/main.js

```diff
@@ -0,0 +1,14 @@
+var errorModule = {
+  show: function() {
+    document.getElementById('error').style.display='block'; 
+  },
+  hide: function(){
+    document.getElementById('error').style.display='none'; 
+  }, 
+  change: function(text, showImmediately) {
```

Comment:
I'm not sure `change` is that useful — I would wrap it up into `show`, and probably rename `show` to `error`, so you get:

    errorModule.error('ahhhh something bad happened');

Or even,

    Errors.error('omg badness');

This way, the code that uses errors doesn't really have to care about how or when they're displayed. If you want to give the caller more control, the typical pattern is to allow them to pass in options:

   Errors.error("this app sure does fail a lot", {show: 'forever'});

Example of usage:

   SSN.valid(ssn.value) || Errors.error('Enter a valid SSN', {show: 'forever', forNode: ssn});